R/loadReg2.R

```diff
@@ -323,42 +324,66 @@ predictSolute.loadReg2 <- function(
   # default. nchunks is the number of chunks required.
   chunk.size <- 176000
   nchunks <- ceiling(nrow(newdata) / chunk.size)
-  newdata <- lapply(1:nchunks, function(i) {
+  datachunks <- lapply(1:nchunks, function(i) {
     newdata[((i-1)*chunk.size + 1):min(i*chunk.size, nrow(newdata)),]
   })
   
   # Now do the prediction for each chunk and then reassemble the full set of
   # predictions
-  preds_raw <- lapply(newdata, function(datachunk) {
+  preds_lin_raw <- lapply(datachunks, function(datachunk) {
     switch(
       flux.or.conc,
-      "flux"=predLoad(
+      "flux"={
         # ... may contain load.units, seopt, print
-        fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level, ...), 
-      "conc"=predConc(
-        fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level) 
+        predLoad(fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level, ...)
+      }, 
+      "conc"={
+        predLoad_args <- c('load.units','seopt','print')[(c('load.units','seopt','print') %in% names(list(...)))]
+        if(length(predLoad_args) > 0) warning(paste("these args are ignored for flux.or.conc='conc':", paste(predLoad_args, collapse=', ')))
+        predConc(fit=load.model@fit, newdata=datachunk, by="unit", allow.incomplete=FALSE, conf.int=level) 
+      }
     )
   })
-  preds_raw <- do.call(rbind, preds_raw)
+  preds_lin_raw <- do.call(rbind, preds_lin_raw)
+  preds_col <- .sentenceCase(flux.or.conc)
+  
+  # Generate predictions in log space in case these were requested. Uses the 
+  # assumption, as in predLoad and predConc, that the Ferguson (1986) method is
+  # adequate for re-retransformation. (predLoad and predConc use these to
+  # compute prediction intervals)
+  KDays <- !is.na(preds_lin_raw[["SEP"]])
+  preds_log_raw <- linToLog(meanlin=preds_lin_raw[[preds_col]][KDays], sdlin=preds_lin_raw$SEP[KDays])
   
   # Format predictions; add intervals if requested
-  preds_col <- .sentenceCase(flux.or.conc)
   if(interval == "none") {
-    preds <- formatPreds(preds_raw[[preds_col]], 
-                         from.format=flux.or.conc, to.format=flux.or.conc, 
-                         newdata=NULL, metadata=metadata, attach.units=attach.units)
+    preds <- switch(
```

Comment:
There's nested logic in all three functions that can be done with switch or if statements, or a combination of the two. When you're choosing between just two options (e.g., lin or log) there's not much difference between switch and if, and I think my cleanliness sensibility is still evolving on this topic. I think switch statements are cleaner when they allow me to get the important information all grouped together, prominent, and vertically aligned - e.g., in this case 'linear' and 'log' are the start (prominent and aligned) of two adjacent lines (grouped) - so that it's easier to see quickly what choice is being made and what the consequences are. That alignment and grouping breaks down a bit when you have larger chunks of code within each option, in which case sometimes I go back to if statements because they use up less whitespace on the left...but at that point I usually don't feel strongly either way.